plugins/modules/notification/discord.py

```diff
@@ -0,0 +1,168 @@
+    type: list
+    elements: str
+'''
+
+EXAMPLES = """
+- name: Send a message to the discord channel
+  community.general.discord:
+    webhook_id: "00000"
+    webhook_token: "XXXYYY"
+    content: "This is a message from ansible"
+
+- name: Send a message to the discord channel with specific username and avatar
+  community.general.discord:
+    webhook_id: "00000"
+    webhook_token: "XXXYYY"
+    content: "This is a message from ansible"
+    username: Ansible
+    avatar_url: "https://docs.ansible.com/ansible/latest/_static/images/logo_invert.png"
+
+- name: Send a embedded message to the discord channel
+  community.general.discord:
+    webhook_id: "00000"
+    webhook_token: "XXXYYY"
+    embeds:
+      - title: "Embedded message"
+        description: "This is an embedded message"
+        footer:
+          text: "Author: Ansible"
+        image:
+          url: "https://docs.ansible.com/ansible/latest/_static/images/logo_invert.png"
+"""
+
+RETURN = """
+http_code:
+  description:
+    - Response Code returned by discord API.
+  returned: always
+  type: int
+  sample: 204
+"""
+
+from ansible.module_utils.urls import fetch_url
+from ansible.module_utils.basic import AnsibleModule
+
+
+def discord_text_msg(module):
+
+    webhook_id = module.params['webhook_id']
+    webhook_token = module.params['webhook_token']
+    content = module.params['content']
+    user = module.params['username']
+    avatar_url = module.params['avatar_url']
+    tts = module.params['tts']
+    embeds = module.params['embeds']
+
+    headers = {
+        'content-type': 'application/json'
+    }
+
+    url = "https://discord.com/api/webhooks/%s/%s" % (
+        webhook_id, webhook_token)
+
+    payload = {
+        'content': content,
+        'username': user,
+        'avatar_url': avatar_url,
+        'tts': tts,
+        'embeds': embeds,
+    }
+
+    payload = module.jsonify(payload)
+
+    response, info = fetch_url(module, url, data=payload, headers=headers)
+    return response, info
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            webhook_id=dict(type='str', required=True),
+            webhook_token=dict(type='str', required=True, no_log=True),
+            content=dict(type='str'),
+            username=dict(type='str'),
+            avatar_url=dict(type='str'),
+            tts=dict(type='bool', default=False),
+            embeds=dict(type='list', elements='str'),
+        ),
+        required_one_of=[['content', 'embeds']],
+        supports_check_mode=False
+    )
+
+    result = dict(
+        changed=False,
+        http_code='',
+    )
+
+    if module.check_mode:
+        module.exit_json(changed=False)
+    else:
+        response, info = discord_text_msg(module)
+
+    if info['status'] != 204:
+        result['http_code'] = info['status']
+        result['msg'] = info['msg']
+        result['failed'] = True
+        result['response'] = response
+        module.fail_json(**result)
```

Comment:
```suggestion
        module.fail_json(http_code=info['status'], msg=info['msg'], response=response)
```

Reply:
Given that the token is being appended to the URL itself (line 111), maybe we should not return the response object, as we might be leaking the token to logs.

Reply:
I checked this. For the token no_log is enabled. In the log the token is masked in the URL. Example: `https://discord.com/api/webhooks/0000000/**************`